polus-label-to-vector-plugin/src/main.py

```diff
@@ -1,190 +1,378 @@
+    """
+    function = dynamics.masks_to_flows_gpu if use_gpu else dynamics.masks_to_flows_cpu
+
+    if numpy.any(mask_slice):
+        shape = numpy.shape(mask_slice)
+        labels, mask_slice = numpy.unique(mask_slice, return_inverse=True)
+        mask_slice = mask_slice.reshape(shape)
+        if len(labels) == 1:
+            mask_slice += 1
+        flow_subset = function(mask_slice, device=device)[0]
+    else:
+        flow_subset = numpy.zeros_like(mask_slice, dtype=numpy.float32)
+
+    return flow_subset
+
+
+def flow_thread_3d(
+        input_path: Path,
+        zarr_path: Path,
+        use_gpu: bool,
+        device: torch.device,
+        coordinates: tuple[int, int],
+) -> bool:
+    """ Converts labels to flows.
 
     This function converts labels in each tile to vector field.
 
     Args:
-        input_path(path): Path of input image collection
-        zfile(path): Path where output zarr file will be saved
-        x(int): Start index of the tile in x dimension of image
-        y(int): Start index of the tile in y dimension of image
-        z(int): Z slice of the  image
+        input_path: Path of input image collection.
+        zarr_path: Path where output zarr file will be saved.
+        use_gpu: Path where output zarr file will be saved.
+        device: Path where output zarr file will be saved.
+        coordinates: (x, y) coordinates of the tile in the image.
 
     """
-    
+
     logging.basicConfig(format='%(asctime)s - %(name)-8s - %(levelname)-8s - %(message)s',
                         datefmt='%d-%b-%y %H:%M:%S')
-    logger = logging.getLogger("flow")
-    logger.setLevel(logging.INFO)
+    flow_logger = logging.getLogger("flow_3d")
+    flow_logger.setLevel(logging.DEBUG)  # TODO: Change back to INFO
+
+    # noinspection PyTypeChecker
+    root = zarr.open(str(zarr_path))[0]
+    x, y = coordinates
+
+    try:
+        with BioReader(input_path) as reader:
+            x_min = max(0, x - TILE_OVERLAP)
+            x_max = min(reader.X, x + TILE_SIZE + TILE_OVERLAP)
+            y_min = max(0, y - TILE_OVERLAP)
+            y_max = min(reader.Y, y + TILE_SIZE + TILE_OVERLAP)
+
+            tile = reader[y_min:y_max, x_min:x_max, 0:reader.Z, 0, 0].squeeze()
+            tile = tile.transpose(2, 0, 1)
+
+            if not numpy.any(tile):
+                flow_logger.debug(f'got tile with no objects in file {input_path.name}')
+                flow_shape = (3, reader.Z, y_max - y_min, x_max - x_min)
+                flow = numpy.zeros(flow_shape, dtype=numpy.float32)
+            else:
+                # Normalize
+                labels, masks = numpy.unique(tile, return_inverse=True)
+                if len(labels) == 1:
+                    flow_logger.debug(f'got tile with only one object in file {input_path.name}')
+                    masks += 1
+
+                masks = masks.reshape(reader.Z, y_max - y_min, x_max - x_min)
```

Comment:
Isn't this code more or less a replicate of the `clean_3d_flow`? Maybe you could include a `bool` in the return from `clean_3d_flow` that tells you if you get any objects in the image.